File: mera-code/0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (c == '(' || c == '[' || c == '{')
                stack.push(c);
            else {
                if (stack.isEmpty())
                    return false;
                if (!isMatching(stack.pop(), c))
                    return false;
            }
        }
        return stack.isEmpty();
    }

    boolean isMatching(char c1, char c2) {
        if ((c1 == '(' && c2 == ')') ||
                (c1 == '{' && c2 == '}') ||
                (c1 == '[' && c2 == ']')) {
            return true;
        }
        return false;
    }
}

File: mera-code/0020-valid-parentheses/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    for(let i=0; i<s.length; i++) {
        if(stack.length > 0 && validatePair(stack[stack.length-1], s.charAt(i)))
            stack.pop();
        else
            stack.push(s.charAt(i));
    }
    return stack.length === 0;
};

function validatePair(c1, c2) {
    if(c1 === '(' && c2 === ')')
        return true;
    else if(c1 === '[' && c2 === ']')
        return true;
    else if(c1 === '{' && c2 === '}')
        return true;
    else 
        return false;
}
